### DISCUSSION TITLE
How to do transaction for Custom Token.

### DISCUSSION THREAD

**Discussion by princeSingh008** [WalletConnect/WalletConnectUnity#159] [Q&A]
Hi,
I am exploring this plugin for my project. I want to use custom token deployed in the blockchain. I want to know can I use this plugin for custom token? Please help me for using the custom token.

**Comment by skibitsky** [comment]
Hi,

Yes, there is an example in the documentation:

https://docs.walletconnect.com/advanced/walletconnectmodal/usage?platform=unity#interaction-with-smart-contracts

**Reply by princeSingh008** [reply]
Hi, Thanks for the response but I was unable to find that how can I use custom ABI for custom function calling. I want to call ERC20 Read/Write function. Can you please share any screenshot or example for that? This plugin will be perfect if I got to know how to interact with custom token.

**Reply by skibitsky** [reply]
The **Interaction with Smart Contracts** section shows how to make transaction against `approve` method of ERC-20 token.

Please note, that WalletConnectModal only handles wallet connection and signing. To build requests and use contracts ABI you need to use Nethereum. Our example doesn't use Nethereum's ABI tools for simplicity. Please refer to [Nethereum ABI encoding documentation](https://docs.nethereum.com/en/latest/nethereum-abi-encoding/) for more details.
